public/app/plugins/datasource/jaeger/datasource.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  dateMath,
+  DateTime,
+  MutableDataFrame,
+  DataSourceApi,
+  DataSourceInstanceSettings,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataQuery,
+} from '@grafana/data';
+import { BackendSrv, DatasourceRequestOptions } from 'app/core/services/backend_srv';
+import { TimeSrv } from 'app/features/dashboard/services/TimeSrv';
+
+import { Observable, from, of } from 'rxjs';
+
+function serializeParams(data: Record<string, any>) {
+  return Object.keys(data)
+    .map(k => `${encodeURIComponent(k)}=${encodeURIComponent(data[k])}`)
+    .join('&');
+}
+
+export type JaegerQuery = {
+  query: string;
+} & DataQuery;
+
+export class JaegerDatasource extends DataSourceApi<JaegerQuery> {
+  /** @ngInject */
+  constructor(
+    private instanceSettings: DataSourceInstanceSettings,
+    private backendSrv: BackendSrv,
```

Comment:
Instead of DI you can:
- import {getBackendSrv} from '@grafana/runtime'
- import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';